File: trustly-android-sdk/src/androidTest/java/net/trustly/android/sdk/views/oauth/TrustlyOAuthViewTest.java
package net.trustly.android.sdk.views.oauth;

import static org.junit.Assert.assertNotNull;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.MediumTest;

import net.trustly.android.sdk.views.TrustlyViewTest;

import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
@MediumTest
public class TrustlyOAuthViewTest extends TrustlyViewTest {

    @Test
    public void shouldValidateTrustlyOAuthViewInstance() {
        scenario.onActivity(activity -> {
            TrustlyOAuthView trustlyOAuthView = new TrustlyOAuthView(activity.getApplicationContext());
            assertNotNull(trustlyOAuthView);
        });
    }

    @Test
    public void shouldValidateTrustlyOAuthViewGetWebViewMethod() {
        scenario.onActivity(activity -> {
            TrustlyOAuthView trustlyOAuthView = new TrustlyOAuthView(activity.getApplicationContext());
            assertNotNull(trustlyOAuthView.getWebView());
        });
    }

}

File: trustly-android-sdk/src/androidTest/java/net/trustly/android/sdk/views/oauth/TrustlyOAuthViewTest.kt
package net.trustly.android.sdk.views.oauth

import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.MediumTest
import net.trustly.android.sdk.TrustlyActivityTest
import net.trustly.android.sdk.interfaces.TrustlyEvents
import net.trustly.android.sdk.mock.MockActivity
import net.trustly.android.sdk.views.TrustlyView
import net.trustly.android.sdk.views.events.TrustlyEventsImpl
import org.junit.Assert.assertNotNull
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@MediumTest
class TrustlyOAuthViewTest : TrustlyActivityTest() {

    private lateinit var trustlyEvents: TrustlyEvents

    @Before
    override fun setUp() {
        super.setUp()

        trustlyEvents = TrustlyEventsImpl
    }

    @Test
    fun shouldValidateTrustlyOAuthViewInstance() {
        scenario.onActivity { activity: MockActivity ->
            val trustlyOAuthView = TrustlyOAuthView(
                activity.applicationContext,
                TrustlyView(activity.applicationContext),
                trustlyEvents
            )
            assertNotNull(trustlyOAuthView)
        }
    }

    @Test
    fun shouldValidateTrustlyOAuthViewGetWebViewMethod() {
        scenario.onActivity { activity: MockActivity ->
            val trustlyOAuthView = TrustlyOAuthView(
                activity.applicationContext,
                TrustlyView(activity.applicationContext),
                trustlyEvents
            )
            assertNotNull(trustlyOAuthView.webView)
        }
    }

}